03week/towersOfHanoi.js

```diff
@@ -13,34 +21,86 @@ let stacks = {
   c: []
 };
 
+let moveCount = 0;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // You can only move the top number (pop method) of any stack, and you can only move one number at a time, and you move the number the another stack with the push method.
+  // const userObj = stack[].pop();
+  if (isValid(startStack, endStack) && isLegal(startStack, endStack)) {
+    stacks[endStack].push(stacks[startStack].pop());
+    moveCount++;
+  } else {
+    console.log("Please try again!");
+    return false;
+  }
 }
 
-function isLegal() {
-  // Your code here
+function isValid(startStack, endStack) {
+  if ((startStack === "a" || startStack === "b" || startStack === "c") && (endStack === "a" || endStack === "b" || endStack === "c")) {
+    return true;
+  } else {
+    console.log("This is an invalid entry, please try again!");
+    return false;
+  }
+}
 
+// function isLegal(startStack, endStack) {
+  // You can only move numbers to either empty rows, or on top of a larger numbers. if the last item of the array that you are moving from is less than the last item of the array you are moving to.
+  // You can only move the numbers that are part of the game, in this case 1 - 4.
+  // if (stacks[endStack].length === 0) {
+  //   return true;
+  // } else if (stacks[startStack].pop() < stacks[endStack].pop()){ This is my code and I know it is what is messing up my game play but I do not understand why. Below is something I got from someone else
+//   } else if (stacks[startStack][(stacks[startStack].length) - 1] < stacks[endStack][(stacks[endStack].length) - 1]) {
+//     return true;
+//   } else if (stacks[startStack].length === 0) {
+//     console.log("This stack is empty, please choose another!");
+//     return false;
+//   } else {
+//     console.log("That is an illegal move, please try again!");
+//     return false;
+//   }
+// }
+
+function isLegal(startStack, endStack) {
+  if (stacks[startStack].length === 0) {
```

Comment:
This function organization is off and you shouldn't return true or false explicitly. Here's how this should be organized. 
```
if (!stacks[startStack].length === 0) {
     return (stacks[endStack].length === 0) || (stacks[startStack][(stacks[startStack].length) - 1] < stacks[endStack][(stacks[endStack].length) - 1])
}
```